smart-storage-unit/packages/contracts/src/systems/SmartStorageUnitSystem.sol

```diff
@@ -87,17 +87,14 @@ contract SmartStorageUnitSystem is System {
       quantity
     );
 
+    uint256 calculatedInput = quantity-quantityInputItemLeftOver;
+
     uint256 itemObjectIdOut = RatioConfig.getItemOut(smartObjectId, inventoryItemIdIn);
 
     TransferItem[] memory inItems = new TransferItem[](1);
-    inItems[0] = TransferItem(inventoryItemIdIn, ssuOwner, quantity);
+    inItems[0] = TransferItem(inventoryItemIdIn, ssuOwner, calculatedInput);
 
     TransferItem[] memory ephTransferItems = new TransferItem[](1);
-    ephTransferItems[0] = TransferItem(itemObjectIdOut, ssuOwner, quantityInputItemLeftOver);
```

Comment:
can you explain whats the bug here and how this change fixes the bug ?

Reply:
It was due to the way that items were being calculated potentially - but doing a different method lets us simplify it to less transfers :)

Reply:
inItems is the quantity the user wants to drop in the inventory, I dont understand why that needs to be changes or calculated